src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Services/BlobCacheServiceTests.cs

```diff
@@ -57,8 +50,13 @@ public async Task SetItemAsync()
 
             var blobStorageService = new Mock<IBlobStorageService>(MockBehavior.Strict);
 
-            blobStorageService.Setup(mock =>
-                    mock.UploadAsJson(PublicContent, cacheKey.Key, "test item", null))
+            blobStorageService.Setup(mock => mock.UploadAsJson(
```

Comment:
Would this be worth adding to MockBlobStorageServiceExtensions?

Reply:
Good idea, made this change.

Reply:
Adding the extension method to be used here is fine, and that change is in, but I was just looking at whether this new method could also be used to replace setup behaviour in `PermalinkServiceTests`.

There `path` and `content` are matchers like `It.Is<string>(path => path.EndsWith(".json.zst"))` and `It.IsAny<PermalinkTableViewModel>()`.

It seems like we can't do this, although passing a matcher will happily compile.

The only way I could get this to work would be with an extra extension method, something like this:

```
        public static IReturnsResult<IBlobStorageService> SetupUploadAsJson<T>(
            this Mock<IBlobStorageService> service,
            IBlobContainer container,
            Expression<Func<string, bool>> pathMatcher,
            Expression<Func<T, bool>> contentMatcher,
            string? contentEncoding = null,
            JsonSerializerSettings? settings = null,
            CancellationToken cancellationToken = default)
        {
            return service.Setup(s =>
                    s.UploadAsJson(container,
                        It.Is(pathMatcher),
                        It.Is(contentMatcher),
                        contentEncoding,
                        settings,
                        cancellationToken))
                .Returns(Task.CompletedTask);
```

This can be used something like this

```
            blobStorageService.SetupUploadAsJson<PermalinkTableViewModel>(
                    container: BlobContainers.PermalinkSnapshots,
                    pathMatcher: path => path.EndsWith(".json.zst"),
                    contentMatcher: content => true,
                    contentEncoding: ContentEncodings.Zstd)
```

This works, but I decided to leave it as it is for now. Just wondering if you have an opinion on this or know of any cleaner way to do this?

Reply:
As per chat yesterday, seems to be all down to Expression parsing in Moq and not much we can do, so it's a shame but it's all good the way it is for now thanks.